compiler/msgs.nim

```diff
@@ -134,7 +134,7 @@ type
     hintProcessing, hintCodeBegin, hintCodeEnd, hintConf, hintPath,
     hintConditionAlwaysTrue, hintName, hintPattern,
     hintExecuting, hintLinking, hintDependency,
-    hintSource, hintPerformance, hintStackTrace, hintGCStats,
+    hintSource, hintSourceError, hintPerformance, hintStackTrace, hintGCStats,
```

Comment:
Why do we need ``hinSourceError`` when ``hinSource`` already exists and was used for the same?

Reply:
after this PR:
hintSourceError only prints source code for errors
hintSource prints source code for all entries

hintSourceError only adds 2 lines in compilation messages (including caret `^`), because there can only be 1 error (IUUC) eg:
```
/Users/timothee/git_clone/nim/timn/tests/t25_hint_source.nim(6, 3) Error: undeclared identifier: 'a'
    a: int
    ^
```
so it's ok to have it on by default with default verbosity 1 (as done with this PR)

hintSource adds more lines to compilation messages, so I only enable it for `hints[source]`
```
/Users/timothee/git_clone/nim/timn/tests/t25_hint_source.nim(2, 39) Warning: expr is deprecated [Deprecated]
  macro foo1*(cond: untyped, msg = ""): expr = discard
                                        ^
/Users/timothee/git_clone/nim/timn/tests/t25_hint_source.nim(3, 39) Warning: expr is deprecated [Deprecated]
  macro foo2*(cond: untyped, msg = ""): expr = discard
                                        ^
/Users/timothee/git_clone/nim/timn/tests/t25_hint_source.nim(6, 3) Error: undeclared identifier: 'a'
    a: int
    ^
```